sklearn/datasets/samples_generator.py

```diff
@@ -1351,6 +1351,42 @@ def make_s_curve(n_samples=100, noise=0.0, random_state=None):
 
     return X, t
 
+def make_trefoil_knot(n_samples=100, noise=0.0, random_state=None):
+    """Generate an Uniform Trefoil Knot 
+
+    Read more in the :ref:`User Guide <sample_generators>`.
```

Comment:
I have detailed this function from this pull request -- I will make a new pull request with better documentation and an example